5118 down arrow to select elements

panacotar opened the pull request:
<!--Read comments, before committing pull request read checklist again

# Checklist:

- I have performed a self-review of my own code,
- I have commented my code, particularly in hard-to-understand areas,
- I have made corresponding changes to the documentation,
- I have added tests that prove my fix is effective or that my feature works,
- New and existing unit tests pass locally with my changes ("bundle exec rake"),
- Title include "WIP" if work is in progress.
- I acknowledge that I will *not* force push my branch once reviews have started.

-->

Resolves #5118  <!--fill issue number-->

### Description
<!-- Please include a summary of the change and which issue is fixed. 
Please also include relevant motivation and context.
Guide questions:
  - What motivated this change (if not already described in an issue)?
  - What alternative solutions did you consider?
  - What are the tradeoffs for your solution?
   
List any dependencies that are required for this change. (gems, js libraries, etc.)

Include anything else we should know about. -->

There are two types of select elements here:
- f.input (collections)
- select2 JS

For the first one, the majority of the inputs are wrapped in `input-group`. So I decided to extend the `form-select` class for those. This applies to all `select` elements wrapped in `input-group`.
I tried to handle it in the config file > [input_group](https://github.com/rubyforgood/human-essentials/blob/d1c52ea616c51e5bcea020e88e9cc7d49a746078/config/initializers/simple_form_bootstrap.rb#L375). But without success, adding `form-select` class here will add it to all inputs, I couldn't find a way to add it only to the `select` tags.

For select2 JS, I appended the `form-select` class in the select2 controller during the component initialization.

### Type of change

<!-- Please delete options that are not relevant. -->

* Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?

<!-- Please describe the tests that you ran to verify your changes. 
Provide instructions so we can reproduce. 
Do we need to do anything else to verify your changes? 
If so, provide instructions (including any relevant configuration) so that we can reproduce? -->

### Screenshots
<!--Optional. Delete if not relevant. 
Include screenshots (before / after) for style changes, highlight edited element.-->

- The form in the issue - /manage/edit

![5118](https://github.com/user-attachments/assets/6cc5eaf7-ff44-413d-aaaa-bffd4c800259)

- The New Product Drive form, under /product_drives/new

![5118-1](https://github.com/user-attachments/assets/0c115f61-17a9-4632-8d9a-bf5c71cfbc1e)

cielf reviewed (dismissed):
Hey @panacotar -- thanks for taking this on.

Alas,  your solution has side effects -- we end up with doubled arrows.  See below (this is on the distributions, but the same would occur anywhere we have items, at least,  which is most of the forms in the system)

![Screenshot 2025-03-31 at 10 04 21 AM](https://github.com/user-attachments/assets/f00d00df-74c8-4b49-b49d-d8d1ff14f07e)

@dorner -- do you have insight on our best approach here? -- I like the idea of having all the drop-downs have the same mechanism, but I don't know if there are other considerations.

panacotar commented:
> Hey @panacotar -- thanks for taking this on.
> 
> Alas, your solution has side effects -- we end up with doubled arrows. See below (this is on the distributions, but the same would occur anywhere we have items, at least, which is most of the forms in the system)
> ...
> 
> @dorner -- do you have insight on our best approach here? -- I like the idea of having all the drop-downs have the same mechanism, but I don't know if there are other considerations.

You are right, I missed that.
I updated it by removing the `form-select` class from select2 JS elements. Some inputs were still missing the arrow (those wrapped in `form-group`), so I added that CSS rule as 

On the issue, the *Partner Profile Sections* input is a select2 **multi-box**. I noticed that it is meant for multi-box selects to be without the arrow, you can see in the official documentation:
https://select2.org/getting-started/basic-usage#multi-select-boxes-pillbox

We can still add an arrow to it using CSS. Here is an example how:
https://stackoverflow.com/a/46363785/13257655

cielf commented:
Hey @pancotar -- I did a pass through the app to check everything,  and I noticed a couple of other instances. -- specifically that when you do a new 'quantity' request on behalf of a partner,  neither the drop-down for selecting the partner nor the drop-down for selecting the item have down arrows.    I think this is pre-existing,  and it's technically out of scope of what was asked for  --- would you like to add it in,  or should we make up a new issue for it (to see what I mean:  sign in as org_Admin1@example.com,  select "requests"  then "New Quantity Request".

panacotar commented:
> Hey @pancotar -- I did a pass through the app to check everything, and I noticed a couple of other instances. -- specifically that when you do a new 'quantity' request on behalf of a partner, neither the drop-down for selecting the partner nor the drop-down for selecting the item have down arrows. I think this is pre-existing, and it's technically out of scope of what was asked for --- would you like to add it in, or should we make up a new issue for it (to see what I mean: sign in as [org_Admin1@example.com](mailto:org_Admin1@example.com), select "requests" then "New Quantity Request".

I will add these fixes as well. The original 5118 issue asked for all dropdowns to be fixed, so it falls under this.
Yes, it is pre-existing, I went back through commits to detect where it was introduced. It seems it has been like this for a while.
Pushing my fixes soon.

panacotar commented:
I added the missing arrow to 3 more selects:
- new request modal
- item select at */partners/individuals_requests/new*
- item select at */partners/requests/new*

cielf commented:
LGTM -- over to @dorner for any technical concerns.
